api/src/org/apache/cloudstack/api/command/user/snapshot/CreateSnapshotCmd.java

```diff
@@ -224,8 +224,8 @@ public void execute() {
 
         Snapshot.LocationType locationTypeToReturn = Snapshot.LocationType.values()[0];
 
-        if (locationType != null && locationType >= 0 && locationType < Snapshot.LocationType.values().length) {
-            locationTypeToReturn = Snapshot.LocationType.values()[locationType];
+        if (locationType != null && locationType >= 0 && locationType <= Snapshot.LocationType.values().length) {
```

Comment:
I thought I fixed that. I did exactly what you decribed here Mike. It looke like somehow it got missed. I will fix that.